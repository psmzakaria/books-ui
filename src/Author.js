import React, { Component } from 'react';
const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";
class Author extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }
  componentDidMount() {
    this.getAuthors("");
  }
  getAuthors = async search => {
    const url = `${API_HOST}/authors`;
    const response = await fetch(url);
    const authorsData = await response.json();
    this.setState({ authors: authorsData });
  };

  render() {
    return (
      <div>
        {this.state.authors.map((author, i) => {
          return <li> {author.name}</li>;
        })}
      </div>
    );
  }
}

export default Author;
