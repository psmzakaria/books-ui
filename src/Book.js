import React, { Component } from "react";
const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";
class Book extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
  componentDidMount() {
    this.getBooks("");
  }

  getBooks = async search => {
    const url = `${API_HOST}/books`;
    // const response = await fetch(url);
    // const bookData = await response.json();
    // this.setState({ books: bookData });

    this.setState({
      books: [
        {
          title: "HARRY POTTER",
          _id: "23094823048"
        }
      ]
    });
  };

  render() {
    return (
      <div>
        {this.state.books.map((book) => {
          return <li key ={book._id}> {book.title}</li>;
        })}
      </div>
    );
  }
}

export default Book;
