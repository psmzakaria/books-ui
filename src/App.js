import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import Book from "./Book";
import Author from "./Author";
const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";
class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return (
      <div>
        <p>Books</p>
        <Book />
        <p>Authors</p>
        <Author />
      </div>
    );
  }
}

export default App;
