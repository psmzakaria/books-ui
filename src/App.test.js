import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import Book from "./Book";
import Author from "./Author";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the book component of the app ", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Book />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});

it("renders the Authors component of the app ", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Author />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
