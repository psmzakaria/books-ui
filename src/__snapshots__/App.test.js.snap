// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Authors component of the app  1`] = `
<div>
  
</div>
`;

exports[`renders the book component of the app  1`] = `
<div>
  
</div>
`;
